### PR TITLE
Fix chat backend

### PR DESCRIPTION
**Pull request by Sheepsody**
Fixed the issues with timestamps in the backend

### CODE REVIEW

File: chat_server/index.ts [hunk 1]
```diff
@@ -48,7 +37,10 @@ io.use(socketio_jwt.authorize(jwtOption));
 
 // Main Socket
 io.sockets.on("connection", (socket) => {
+  console.log("New user connected !");
```
Comment (Piwakk) [inline chat_server/index.ts:40]: Remove `console.log`

File: chat_server/index.ts [hunk 2]
```diff
@@ -48,7 +37,10 @@ io.use(socketio_jwt.authorize(jwtOption));
...
+
   socket.on("message", async (request: any) => {
+    console.log("Received new message : %s", request.message);
```
Comment (Piwakk) [inline chat_server/index.ts:43]: Remove `console.log`

File: chat_server/index.ts [hunk 3]
```diff
@@ -2,34 +2,23 @@ const express = require("express");
...
-    this.posted_on = posted_on;
-  }
-}
-
-/**
- * The JWT authentication is made with https://github.com/auth0-community/auth0-socketio-jwt
- */
-
 const app = express();
 let port = process.env.PORT || 3001;
```
Comment (Piwakk) [inline chat_server/index.ts:12]: Use `const` whenever possible

File: chat_server/index.ts [hunk 4]
```diff
@@ -39,7 +28,7 @@ if (public_key === undefined) {
 
 // Authentification using handshake
 let jwtOption = {
```
Comment (Piwakk) [inline chat_server/index.ts:30]: Use `const` whenever possible

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 node_modules
+package-lock.json
```
Comment (Piwakk) [inline .gitignore:2]: The `package-lock.json` file should be committed

File: chat_server/db.ts [hunk 6]
```diff
@@ -30,12 +35,18 @@ pool
...
-    console.error("❌ Failed to create schema ", err);
+    console.error("Failed to create schema ❌", err);
   });
 
 // Query functions
-const add = async function add(username: string, message: string) {
-  await pool.query(add_query, [username, message]);
+const add = async function add(message: Message) {
+  // Careful: Node-postgres converts the dates to PostgresSQL to match time zone
+  // https://node-postgres.com/features/types
```
Comment (abocquet) [inline chat_server/db.ts:44]: What do we do to avoid this?

Comment (Sheepsody) [reply chat_server/db.ts:44]: Removed the mis-leading comment

Comment (Sheepsody) [reply chat_server/db.ts:44]: Removed the mis-leading comment

File: backend/courses/migrations/0001_initial.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
+# Generated by Django 2.2.10 on 2020-05-25 13:56
+
```
Comment (abocquet) [inline backend/courses/migrations/0001_initial.py:2]: Don't commit migration files. Also this doesn't have  anything to do in the `fix_chat` PR

File: backend/courses/migrations/0002_auto_20200529_1652.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 2.2.10 on 2020-05-29 16:52
+
```
Comment (abocquet) [inline backend/courses/migrations/0002_auto_20200529_1652.py:2]: Same comment

File: backend/courses/migrations/0003_auto_20200530_1206.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.10 on 2020-05-30 12:06
+
+from django.db import migrations, models
```
Comment (abocquet) [inline backend/courses/migrations/0003_auto_20200530_1206.py:3]: Same

File: backend/courses/migrations/0004_auto_20200530_1209.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.10 on 2020-05-30 12:09
+
+from django.db import migrations, models
```
Comment (abocquet) [inline backend/courses/migrations/0004_auto_20200530_1209.py:3]: Same

File: chat_server/package.json [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    "@types/jsonwebtoken": "^8.3.5",
+    "@types/mocha": "^5.2.7",
+    "@types/pg": "^7.11.2",
+    "@types/socket.io": "^2.1.4",
+    "@types/socketio-jwt": "0.0.0",
+    "chai": "^4.2.0",
+    "jsonwebtoken": "^8.5.1",
+    "mocha": "^6.2.2",
+    "mongodb": "^3.4.1",
+    "mongodb-memory-server": "^6.1.1",
```
Comment (abocquet) [inline chat_server/package.json:24]: Why do we need mongodb ?

Comment (Sheepsody) [reply chat_server/package.json:24]: Deprecated import, fixing it and merging master

**Pull request merged**